lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -86,17 +97,23 @@ def set_host_and_port_and_method_info_on_desc
             end
           end
 
-          def server_options(headers)
+          def grpc_headers(metadata)
+            metadata.reject { |k, v| DT_KEYS.include?(k) }
+          end
```

Comment:
Everything in the metadata that is not New Relic Distributed Tracing related is considered to be reportable metadata related to the gRPC request. The metadata will have a key/value pair for the user agent, and then any specific key/value pairs the client code added to a request.